risc0/zkvm/src/groth16.rs

```diff
@@ -0,0 +1,802 @@
+    pub(crate) const BETA_X2: &str =
+        "6375614351688725206403948262868962793625744043794305715222011528459656738731";
+    pub(crate) const BETA_Y1: &str =
+        "21847035105528745403288232691147584728191162732299865338377159692350059136679";
+    pub(crate) const BETA_Y2: &str =
+        "10505242626370262277552901082094356697409835680220590971873171140371331206856";
+    pub(crate) const GAMMA_X1: &str =
+        "11559732032986387107991004021392285783925812861821192530917403151452391805634";
+    pub(crate) const GAMMA_X2: &str =
+        "10857046999023057135944570762232829481370756359578518086990519993285655852781";
+    pub(crate) const GAMMA_Y1: &str =
+        "4082367875863433681332203403145435568316851327593401208105741076214120093531";
+    pub(crate) const GAMMA_Y2: &str =
+        "8495653923123431417604973247489272438418190587263600148770280649306958101930";
+    pub(crate) const DELTA_X1: &str =
+        "18518940221910320856687047018635785128750837022059566906616608708313475199865";
+    pub(crate) const DELTA_X2: &str =
+        "9492326610711013918333865133991413442330971822743127449106067493230447878125";
+    pub(crate) const DELTA_Y1: &str =
+        "19483644759748826533215810634368877792922012485854314246298395665859158607201";
+    pub(crate) const DELTA_Y2: &str =
+        "21375251776817431660251933179512026180139877181625068362970095925425149918084";
+
+    pub(crate) const IC0_X: &str =
+        "5283414572476013565779278723585415063371186194506872223482170607932178811733";
+    pub(crate) const IC0_Y: &str =
+        "18704069070102836155408936676819275373965966640372164023392964533091458933020";
+    pub(crate) const IC1_X: &str =
+        "4204832149120840018317309580010992142700029278901617154852760187580780425598";
+    pub(crate) const IC1_Y: &str =
+        "12454324579480242399557363837918019584959512625719173397955145140913291575910";
+    pub(crate) const IC2_X: &str =
+        "14956117485756386823219519866025248834283088288522682527835557402788427995664";
+    pub(crate) const IC2_Y: &str =
+        "6968527870554016879785099818512699922114301060378071349626144898778340839382";
+    pub(crate) const IC3_X: &str =
+        "6512168907754184210144919576616764035747139382744482291187821746087116094329";
+    pub(crate) const IC3_Y: &str =
+        "17156131719875889332084290091263207055049222677188492681713268727972722760739";
+    pub(crate) const IC4_X: &str =
+        "5195346330747727606774560791771406703229046454464300598774280139349802276261";
+    pub(crate) const IC4_Y: &str =
+        "16279160127031959334335024858510026085227931356896384961436876214395869945425";
+}
+
+/// Groth16 proof object encoded in big endian
+#[cfg(not(target_os = "zkvm"))]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct Groth16Seal {
+    /// Proof 'a' value
+    pub a: Vec<Vec<u8>>,
+    /// Proof 'b' value
+    pub b: Vec<Vec<Vec<u8>>>,
+    /// Proof 'c' value
+    pub c: Vec<Vec<u8>>,
```

Comment:
There are each fix-sized arrays. It would be better to encode that explicitly by using.